app/models/concerns/foreman_git_templates/orchestration/tftp.rb

```diff
@@ -20,9 +20,9 @@ def generate_pxe_template(kind)
         private
 
         def validate_tftp
-          return super unless host.params['template_url']
           return unless tftp? || tftp6?
           return unless host.operatingsystem
+          return super unless host.params['template_url']
```

Comment:
I think we should make sure to have this as the first line in the method to make sure we don't introduce any regressions.
This concern is included into `Nic::Managed`. Can we add something like

```ruby
return super unless host.is_a?(Host::Managed)
```

Reply:
I've changed it accordingly. Hopefully the complexity of the method does not reach it's limit ;)